src/prostredi/editor/clion/cmake.md

```diff
@@ -0,0 +1,92 @@
+## Co je CMake?
+
+CMake je open-source build systém, který automatizuje proces konfigurace a sestavení softwarových projektů napříč různými platformami.
+
+## Co znamená jednotlivý řádek, když vytvořím projekt v CLionu?
+
+Když vytváříme projekt v CLionu, automaticky se vygeneruje soubor **CMakeLists.txt**, který vypadá následovně:
+
+```cmake
+cmake_minimum_required(VERSION 3.21)
+project(testupr C)
+
+set(CMAKE_C_STANDARD 11)
+
+add_executable(testupr main.c)
+```
+
+```cmake
+cmake_minimum_required(VERSION 3.21)
+```
+
+- Tento řádek určuje minimální požadovanou verzi CMake, která je potřebná pro sestavení projektu. V tomto případě je požadovaná verze 3.21. Pokud máte starší verzi CMake, sestavení projektu nebude
+  možné.
+
+```cmake
+project(testupr C)
+```
+
+- Definuje název projektu, který je v tomto případě "testupr", a určuje, že projekt je napsán v jazyce C.
+
+```cmake
+set(CMAKE_C_STANDARD 11)
+```
+
+- Tento řádek nastavuje verzi standardu jazyka C, kterou má kompilátor použít. Zde je to C11.
+
+```cmake
+add_executable(testupr main.c)
+```
+
+- Tento příkaz říká, že výsledkem buildu bude spustitelný soubor nazvaný "testupr".
+
+> Když chcete přidat další soubory do projektu, jednoduše je přidejte takto:
+> ```cmake
+> add_executable(testupr main.c funkce.c funkce.h)
```

Comment:
```suggestion
> add_executable(testupr main.c funkce.c)
```
`.h` soubory by se do CMake targetů přidávat neměly (protože by se neměly přímo kompilovat). Ty se přeloží tranzitivním vložením z `.c` souborů.

Reply:
Četl jsem, že je lepší je tam přidat, protože některé IDE zobrazuji targety jako složky myslím že VS ? 
Source: https://cliutils.gitlab.io/modern-cmake/chapters/basics.html#making-an-executable

Reply:
Myslím, že ta věta "Targets show up as folders in many IDEs." nesouvisí s tou předchozí větou.

Každopádně CLion ani VSCode by tam hlavičkové soubory neměli potřebovat, tak bych to nechtěl studenty učit, může to pak vypadat, že by je měli přidávat i v command lině, když překládají program ručně.